### DISCUSSION TITLE
how to use the  nginx proxy_pass the docker webui

### DISCUSSION THREAD

**Discussion by honghh2018** [open-webui/open-webui#7366] [General]
Hi developer,
    i want to proxy the webui in nginx with the nginx configure file lying below:
    
#user  nobody;
worker_processes  1;

#error_log  logs/error.log;
#error_log  logs/error.log  notice;
#error_log  logs/error.log  info;

#pid        logs/nginx.pid;

events {
    worker_connections  1024;
}

http {
   
	
	 include       mime.types;
    default_type  application/octet-stream;
    sendfile        on;
    keepalive_timeout  65;
    upstream app-gateway { # 负载均衡
        server 127.0.0.1:3000; # 网关服务的地址和端口
    }
	
	

    #gzip  on;

    #server {
     #   listen       80;
     #   server_name  localhost;

        #charset koi8-r;

        #access_log  logs/host.access.log  main;

    #    location / {
    #        root   html;
    #        index  index.html index.htm;
     #   }

        #error_page  404              /404.html;

        # redirect server error pages to the static page /50x.html
        #
    #    error_page   500 502 503 504  /50x.html;
    #    location = /50x.html {
     #       root   html;
    #    }

        # proxy the PHP scripts to Apache listening on 127.0.0.1:80
        #
        #location ~ \.php$ {
        #    proxy_pass   http://127.0.0.1;
        #}

        # pass the PHP scripts to FastCGI server listening on 127.0.0.1:9000
        #
        #location ~ \.php$ {
        #    root           html;
        #    fastcgi_pass   127.0.0.1:9000;
        #    fastcgi_index  index.php;
        #    fastcgi_param  SCRIPT_FILENAME  /scripts$fastcgi_script_name;
        #    include        fastcgi_params;
        #}

        # deny access to .htaccess files, if Apache's document root
        # concurs with nginx's one
        #
        #location ~ /\.ht {
        #    deny  all;
        #}
    #}

    # another virtual host using mix of IP-, name-, and port-based configuration
    #
    #server {
    #    listen       8000;
    #    listen       somename:8080;
    #    server_name  somename  alias  another.alias;

    #    location / {
    #        root   html;
    #        index  index.html index.htm;
    #    }
    #}

    # HTTPS server
    #
    #server {
    #    listen       443 ssl;
    #    server_name  localhost;

    #    ssl_certificate      cert.pem;
    #    ssl_certificate_key  cert.key;

    #    ssl_session_cache    shared:SSL:1m;
    #    ssl_session_timeout  5m;

    #    ssl_ciphers  HIGH:!aNULL:!MD5;
    #    ssl_prefer_server_ciphers  on;

    #    location / {
    #        root   html;
    #        index  index.html index.htm;
    #    }
    #}
	
	
	
	server {
    listen 8008;
    server_name  localhost;

    # 动态资源代理
    location  ~/app/(.*) {
        proxy_pass http://app-gateway/$1;
		proxy_pass_request_body on; # 开启获取请求体
		proxy_pass_request_headers on; # 开启获取请求头
		proxy_set_header X-Real-IP $remote_addr; # 记录真实发出请求的客户端IP
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; # 记录代理信息
       
    }

    # 静态资源代理
    location /_app {
        proxy_pass http://app-gateway/_app;
        
    }

    location /static {
        proxy_pass http://app-gateway/static;
        
    }

    # WebSocket 代理
    location /ws {
        proxy_pass http://127.0.0.1:3000;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }
}

}

  But the static file can not be found, 404 error
  
![image](https://github.com/user-attachments/assets/e8da6d43-5e08-4ccc-be2f-a5a6346bdb99)

how to fix this error in proxy, any advice would be appreciated
Best,
hanhuihong

**Comment by wixyskywriter** [comment]
I'm not sure if this is the right place to comment but, i have a standalone installation outside docker, and i'm trying to proxy webui using nginx. However i want open-webui to listen only on localhost. i modified the file __init__.py and replaced host: str = "0.0.0.0", with host: str = "127.0.0.1",
This works with open-webui serve but after update the file is overriden. Any idea? can we have the value as environment variable similar to OLLAMA_HOST
